### DISCUSSION TITLE
How can i add new custom field in post settings?

### DISCUSSION THREAD

**Discussion by Gigagom** [PiranhaCMS/piranha.core#1613] [Q&A] [answered]
Hello, i want to add a image gallery to post settings for displaying a galleries on archive page. 
Can i do it with piranha? 
Please show me how to add a custom field to a post settings.

**Comment by tidyui** [comment, marked as answer]
You can tell a `Region` to be rendered in the **settings** panel by using `Display = RegionDisplayMode.Setting`. For more information, take a look at https://piranhacms.org/docs/master/content/regions.

Best regards

**Reply by Gigagom** [reply]
Thanks a lot, now it appears in the settings! 
But how can i get gallery images with call to _api.Archives.GetByIdAsync?
Or I can get data from a custom post field only by iterating over all posts
